packages/cli/homebrew/scripts/prepare.mjs

```diff
@@ -0,0 +1,93 @@
+import url from 'url'
+import crypto from 'crypto'
+import fs from 'fs'
+import path from 'path'
+import { parse } from 'semver'
+import { pipeline } from 'stream/promises'
+
+const __filename = url.fileURLToPath(import.meta.url)
+const __dirname = path.dirname(__filename)
+
+const CLI_ROOT = path.join(__dirname, '..', '..')
+const DIST_ASSETS = path.join(CLI_ROOT, 'dist')
+
+const { description, homepage, name, version } = JSON.parse(
+  fs.readFileSync(path.join(CLI_ROOT, 'package.json'))
+)
+
+const { GITHUB_SHA_SHORT } = process.env
+
+if (!GITHUB_SHA_SHORT) {
+  console.log('Missing GITHUB_SHA_SHORT in environment; exiting')
+  process.exit(1)
+}
+if (parse(version)?.prerelease.length) {
+  console.log('Not on stable release; exiting')
+  process.exit(1)
+}
+
+async function calculateSHA256(fileName) {
+  const hash = crypto.createHash('sha256')
+  hash.setEncoding('hex')
+  await pipeline(fs.createReadStream(fileName), hash)
+
+  return hash.read()
+}
+
+const TEMPLATES = path.join(CLI_ROOT, 'homebrew', 'templates')
+const fileSuffix = '.tar.xz'
+const ARCH_INTEL = 'x64'
+const ARCH_ARM = 'arm64'
+const versionedName = `${name}@${version}`
+const urlPrefix = `https://github.com/celo-org/developer-tooling/releases/download/${encodeURIComponent(
+  versionedName
+)}`
+
+const fileNamePrefix = `celocli-v${version}-${GITHUB_SHA_SHORT}`
+const fileNameMacIntel = `${fileNamePrefix}-darwin-${ARCH_INTEL}${fileSuffix}`
+const fileNameMacArm = `${fileNamePrefix}-darwin-${ARCH_ARM}${fileSuffix}`
+const fileNameLinuxIntel = `${fileNamePrefix}-linux-${ARCH_INTEL}${fileSuffix}`
+const fileNameLinuxArm = `${fileNamePrefix}-linux-arm${fileSuffix}`
+
+async function updateHomebrewFormula() {
+  const templatePath = path.join(TEMPLATES, 'celocli.rb')
```

Comment:
since there is the comment about this expecting yarn tarball to be run first about about throwing at the top here if those files dont exit with clear message "yarn oclif tarballs"